src/widgetastic_patternfly/__init__.py

```diff
@@ -227,39 +230,58 @@ def dismiss(self):
         for message in self.messages:
             message.dismiss()
 
-    def assert_no_error(self):
-        self.logger.info('asserting there are no error messages')
-        for message in self.messages:
-            if message.type not in {'success', 'info', 'warning'}:
-                self.logger.error('%s: %r', message.type, message.text)
-                raise AssertionError('assert_no_error: {}: {}'.format(message.type, message.text))
-            else:
-                self.logger.info('%s: %r', message.type, message.text)
-
-    def assert_message(self, text, t=None, partial=False):
-        log_part = 'partial match of' if partial else ''
+    def match_messages(self, text=None, t=None, partial=False, inverse=False):
+        msg_type = t if isinstance(t, (tuple, list, set, type(None))) else (t, )
+        op = bool if inverse else not_
+        log_part = 'partial' if partial else 'exact'
         if t is not None:
-            self.logger.info('asserting %s the flash message %r of type %r', log_part, text, t)
-        else:
-            self.logger.info('asserting %s the flash message %r', log_part, text)
-        all_messages = self.messages  # Store for logging on exception without querying twice
-        for message in all_messages:
-            if (partial and text in message.text) or (not partial and text == message.text):
-                if t is not None and message.type == t:
-                    return True
-                elif t is None:
-                    return True
+            self.logger.info('%s match of the flash message %r of type %r', log_part, text, t)
         else:
+            self.logger.info('%s match of the flash message %r', log_part, text)
+        matched_messages = []
+        for message in self.messages:
+            if msg_type and op(message.type in msg_type):
+                continue
+
+            if text and op((partial and text in message.text) or
+                           (not partial and text == message.text)):
+                continue
+
+            matched_messages.append(message)
+        return matched_messages
+
+    def assert_no_error(self, wait=0):
+        self.logger.info('asserting there are no error messages')
+        try:
+            msgs = wait_for(self.match_messages, func_kwargs=dict(t={'success', 'info', 'warning'},
```

Comment:
optional:
unpack the tuple on assignment instead of indexing the wait_for return:

`msgs, _ = wait_for(stuff)`